src/keyshare_export.rs

```diff
@@ -0,0 +1,162 @@
+//! Module to export and import key shares, with encryption in transit.
+
+use crate::{errors::Result, keygen::KeySharePrivate};
+use std::{marker::PhantomData, ops::Deref};
+
+mod pke;
+pub use pke::Pke;
+
+mod sodium;
+pub use sodium::SodiumPke;
+
+/// Struct for handling the export and import of KeySharePrivate
+#[derive(Clone, Debug)]
+pub struct KeyShareEncrypted<T: Pke>(Vec<u8>, PhantomData<T>);
+
+impl<T: Pke> From<Vec<u8>> for KeyShareEncrypted<T> {
+    fn from(data: Vec<u8>) -> Self {
+        Self(data, PhantomData)
+    }
+}
+
+impl<T: Pke> From<KeyShareEncrypted<T>> for Vec<u8> {
+    fn from(encrypted: KeyShareEncrypted<T>) -> Vec<u8> {
+        encrypted.0
+    }
+}
+
+impl<T: Pke> Deref for KeyShareEncrypted<T> {
+    type Target = Vec<u8>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl<T: Pke> KeyShareEncrypted<T> {
+    /// Export a `KeySharePrivate` structure by serializing and encrypting it.
+    ///
+    /// # Arguments
+    ///
+    /// * `key_share` - The private key share to export.
+    /// * `receiver_pk` - The public key of the receiver for encryption.
+    /// * `sender_sk` - The private key of the sender for encryption.
+    ///
+    /// # Returns
+    ///
+    /// An encrypted byte vector of the serialized KeySharePrivate.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use tss_ecdsa::keygen::KeySharePrivate;
+    /// use tss_ecdsa::keyshare_export::{KeyShareEncrypted, SodiumPke};
+    /// use sodiumoxide::crypto::box_;
+    ///
+    /// sodiumoxide::init().expect("Failed to initialize sodiumoxide");
+    /// let (sender_pk, sender_sk) = box_::gen_keypair();
+    /// let (receiver_pk, receiver_sk) = box_::gen_keypair();
+    /// let mut rng = rand::thread_rng();
+    /// let key_share = KeySharePrivate::random(&mut rng);
+    ///
+    /// let exported = KeyShareEncrypted::<SodiumPke>::export_keyshare(&key_share, &receiver_pk, &sender_sk)
+    ///     .expect("Failed to export keyshare");
+    /// let exported_bytes: Vec<u8> = exported.into();
+    ///
+    /// let to_import = KeyShareEncrypted::<SodiumPke>::from(exported_bytes);
+    /// let imported = to_import.import_keyshare(&sender_pk, &receiver_sk)
+    ///     .expect("Failed to import keyshare");
+    ///
+    /// assert_eq!(imported, key_share);
+    /// ```
+    pub fn export_keyshare(
```

Comment:
Per my other comment, on removing the generic on the struct. We will then re-introduce the generic here, at the function level: `fn export_keyshare<T: Pke>(...)`.